web_client/panels/DrawWidget.js

```diff
@@ -98,18 +107,34 @@ var DrawWidget = Panel.extend({
     /**
      * Respond to clicking an element type by putting the image
      * viewer into "draw" mode.
+     *
+     * @param {jQuery.Event} [evt] The button click that triggered this event.
+     *      `undefined` to use a passed-in type.
+     * @param {string|null} [type] If `evt` is `undefined`, switch to this draw
+     *      mode.
      */
-    drawElement(evt) {
-        var $el = this.$(evt.currentTarget);
-        var type = $el.data('type');
-        if ($el.hasClass('active')) {
+    drawElement(evt, type) {
+        var $el;
+        if (evt) {
+            $el = this.$(evt.currentTarget);
```

Comment:
For some reason, the tooltips stay open after clicking a button with this change.  Adding `$el.tooltip('hide')` here gets rid of it.